src/Ingest.php

```diff
@@ -133,8 +134,7 @@ private function waitForAcknowledgment($stream): void
             }
 
             $response .= $part;
-
-        } while (strlen($response) < 4);
+        } while (strlen($response) < 4 && ! feof($stream));
```

Comment:
`fread` returns an empty string when the stream reaches its end, rather than returning `false`. This ensure we only attempt to read from an open stream.